### PR TITLE
add commit info for notifications

### PR DESCRIPTION
**Pull request by kim-codefresh**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -956,7 +956,7 @@ func (s *Server) streamApplicationEvents(
 	}
 
 	logCtx.Info("streaming application events")
-	appEvent, err := getApplicationEventPayload(a, es, desiredManifests)
+	appEvent, err := getApplicationEventPayload(a, es, desiredManifests, s, ctx)
```
Comment (roi-codefresh) [inline server/application/application.go]: ctx should always pass as first param

Comment (roi-codefresh) [reply server/application/application.go]: no need to pass the `server`, just make this function a method and use it as receiver type

File: server/application/application.go [hunk 2]
```diff
@@ -1163,15 +1166,21 @@ func getApplicationEventPayload(a *appv1.Application, es *events.EventSource, ma
 		return nil, fmt.Errorf("failed to marshal application event")
 	}
 
+	revisionMetadata, errs := s.RevisionMetadata(ctx, revision)
```
Comment (roi-codefresh) [inline server/application/application.go]: `err` not `errs`

File: server/application/application.go [hunk 3]
```diff
@@ -1148,10 +1148,13 @@ func parseAggregativeHealthErrors(rs *appv1.ResourceStatus, apptree *appv1.Appli
 	return errs
 }
 
-func getApplicationEventPayload(a *appv1.Application, es *events.EventSource, manifestsResponse *apiclient.ManifestResponse) (*events.Event, error) {
+func getApplicationEventPayload(a *appv1.Application, es *events.EventSource, manifestsResponse *apiclient.ManifestResponse, s *Server, ctx context.Context) (*events.Event, error) {
 	obj := appv1.Application{}
 	a.DeepCopyInto(&obj)
 
+	revision := new(application.RevisionMetadataQuery)
+	revision.Revision = &manifestsResponse.Revision
```
Comment (roi-codefresh) [inline server/application/application.go]:
```
revision := &applicationRevisionMetadataQuery{
    Revision: &manifestsResponse.Revision,
}
```

**Comment by noam-codefresh** [comment]
I think this PR should be closed. The relevant fix is in the release branch already, with the message/revision/etc saved on the labels, instead of the source.

**Comment by kim-codefresh** [comment]
> 

you are right the fix was made on https://github.com/codefresh-io/argo-cd/pull/36/commits/0a212f7ae2b247a03cd5fdfac63fdfcbd8512dc6

**Pull request closed without merging**